cmd/agent/app/agent.go

```diff
@@ -316,7 +316,7 @@ func startServiceExportController(ctx controllerscontext.Context) (bool, error)
 	return true, nil
 }
 
-func generateClusterInControllerPlane(opts util.ClusterRegisterOption) (*clusterv1alpha1.Cluster, error) {
+func generateClusterInControllerPlane(opts *util.ClusterRegisterOption) (*clusterv1alpha1.Cluster, error) {
```

Comment:
Generally speaking, using a pointer is cheaper than copy struct, but take this case for example, given this struct is small that means copy struct is still very cheap. The performance improvement is negligible I guess. 

In addition, using a pointer also brings some side effects, for example, how do we know the pointer is not `nil` before we dereferencing it? Should we check it? Or, some static check probably always warns us, that's annoying thing.

Reply:
Got it.  thx!